fairlearn/reductions/_exponentiated_gradient/_lagrangian.py

```diff
@@ -137,11 +140,19 @@ def _call_oracle(self, lambda_vec):
         redW = signed_weights.abs()
         redW = self.n * redW / redW.sum()
 
-        classifier = pickle.loads(self.pickled_estimator)
-        oracle_call_start_time = time()
-        classifier.fit(self.X, redY, sample_weight=redW)
-        self.oracle_execution_times.append(time() - oracle_call_start_time)
-        self.n_oracle_calls += 1
+        redY_unique = np.unique(redY)
+        if len(redY_unique) == 1:
+            logger.debug("redY had single value. Using DummyClassifier")
+            classifier = DummyClassifier(strategy='constant',
+                                         constant=redY_unique[0])
+            classifier.fit(self.X, redY, sample_weight=redW)
```

Comment:
Part of me wants to pull `fit` out of the if statement and add the `fit` time to oracle execution times. If the goal is to track performance, we shouldn't assume that `DummyClassifier` is a no-op.

Reply:
Also--I think that we should always increment `n_oracle_calls`.

Reply:
Modified as discussed

Reply:
is there a reason why not pull out `fit` from the if/then statement?